kotlinx-coroutines-core/common/test/flow/VirtualTime.kt

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2016-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.coroutines
+
+import kotlin.coroutines.*
+import kotlin.jvm.*
+
+private class VirtualTimeDispatcher(enclosingScope: CoroutineScope) : CoroutineDispatcher(), Delay {
```

Comment:
You may be able to hook up to the `TestCoroutineScope` and `TestCoroutineDispatcher` from this PR https://github.com/Kotlin/kotlinx.coroutines/pull/890 instead of using this `VirtualTimeDispatcher` class.

Reply:
I was thinking about it, but at 1.2.1 I am not ready to make `TestCoroutineScope` multiplatform